ext/opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -0,0 +1,49 @@
+import requests
+
+from opentelemetry.sdk.resources import Resource, ResourceDetector
+
+_GCP_METADATA_URL = (
+    "http://metadata.google.internal/computeMetadata/v1?recursive=true"
+)
+_GCP_METADATA_URL_HEADER = {"Metadata-Flavor": "Google"}
+
+
+def get_gce_resources():
+    """ Resource finder for common GCE attributes
+
+        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
+    """
+    # This call will throw if we aren't currently in a GCE instance
+    try:
+        all_metadata = requests.get(
+            _GCP_METADATA_URL, headers=_GCP_METADATA_URL_HEADER
+        ).json()
+    # pylint: disable=broad-except
+    except Exception:
```

Comment:
Ah yeah, so the use case of the two are slightly different. Even if we have raise_on_error = True, we don't want this to raise. The try except here is just for the case we aren't actually in a GCP instance, which the detector should handle  (instead of raising).

Reply:
Shouldn't the detector raise in that case too? It means that it couldn't detect anything. Besides that, you could catch an exception here for reasons other than not running on GCP, like the VM's network is down, the metadata service is down, there is a faulty firewall rule, etc.

Reply:
Hm yeah, that's a good point. The user shouldn't be using the detector when not on GCP anyways. Will update.